### ISSUE TITLE
league/fractal locked @0.16.0 blocking update of other plugin with same dependency

### ISSUE THREAD

**Issue by whoisjuan** [craftcms/element-api#109]
craftcms/element-api has a dependency on fractal. There's a new version  of fractal but although you have it in the schema defined with a caret version range, it's forcing me to have 0.16.0 and not working on 0.18.0. See the following:

  Problem 1
    - rias/craft-scout 1.3.0 requires league/fractal ^0.18.0 -> satisfiable by league/fractal[0.18.0].
    - rias/craft-scout 1.3.0 requires league/fractal ^0.18.0 -> satisfiable by league/fractal[0.18.0].
    - Can only install one of: league/fractal[0.16.0, 0.18.0].
    - Can only install one of: league/fractal[0.18.0, 0.16.0].
    - craftcms/element-api 2.5.4 requires league/fractal ^0.16.0 -> satisfiable by league/fractal[0.16.0].
    - Installation request for craftcms/element-api 2.5.4 -> satisfiable by craftcms/element-api[2.5.4].
    - Installation request for rias/craft-scout 1.3.0 -> satisfiable by rias/craft-scout[1.3.0]

Could we get an update of that dependency requirement?

**Comment by khalwat** [comment]
Here are the Composer requirements:

[craftcms/element-api](https://github.com/craftcms/element-api/blob/v2/composer.json#L29)'s requirements:
```
    "league/fractal": "^0.16.0"
```

[rias/craft-scout](https://github.com/riasvdv/craft-scout/blob/master/composer.json#L26)'s requirements:
```
        "league/fractal": "^0.18.0"
```

These are very compatible semvers... so my guess @whoisjuan it's likely that you have `league/fractal` explicitly listed in your project's main `composer.json` file.

Remove any references to `league/fractal` from your project's `composer.json` and do a `composer update`

**Comment by khalwat** [comment]
Never mind, it's because of the treatment of < 0. releases, so `^0.16.0` will never match anything higher than `0.16.99`

https://semver.mwl.be/#!?package=league%2Ffractal&version=%5E0.16.0&minimum-stability=stable

**Comment by brandonkelly** [comment]
Element API 2.6.0 is out now with Fractal 0.18.

**Comment by jamesmacwhite** [comment]
This seems to be possibly happening again:

```
  Problem 1
    - Installation request for rias/craft-scout ^2.2 -> satisfiable by rias/craft-scout[2.2.0].
    - Conclusion: remove league/fractal 0.18.0
    - Conclusion: don't install league/fractal 0.18.0
    - rias/craft-scout 2.2.0 requires league/fractal ^0.19 -> satisfiable by league/fractal[0.19, 0.19.1, 0.19.2].
    - Can only install one of: league/fractal[0.19, 0.18.0].
    - Can only install one of: league/fractal[0.19.1, 0.18.0].
    - Can only install one of: league/fractal[0.19.2, 0.18.0].
    - Installation request for league/fractal (locked at 0.18.0) -> satisfiable by league/fractal[0.18.0].
```

Scout 2.2 (recently released) is now wanting ^0.19 for league/fractal, where as the Element API is fixed at ^0.18.

**Comment by aaronbushnell** [comment]
Having this same issue where we have package that requires `league/fractal ^0.20.1`. Any shot Element API could get a version bump on this? Looks like the latest version offers a good deal of modern PHP benefits: https://github.com/thephpleague/fractal/releases/tag/0.20
